queries.sql

```diff
@@ -86,4 +86,95 @@ SELECT species, MIN(weight_kg),MAX(weight_kg) FROM animals GROUP BY species;
 -- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
 SELECT species, AVG(escape_attempts)
 FROM animals GROUP BY species
-WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31';
\ No newline at end of file
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31';
+
+--Write queries (using JOIN) to answer the following questions:
+
+
+--What animals belong to Melody Pond?
+SELECT animals.name FROM animals JOIN owners ON animals.owner_id=owners.id WHERE full_name='Melody Pond';
+
+--List of all animals that are pokemon (their type is Pokemon).
+SELECT animals.name FROM animals JOIN species ON animals.species_id=species.id WHERE species_id=1;
+
+--List all owners and their animals, remember to include those that don't own any animal.
+SELECT animals.name,full_name FROM owners LEFT JOIN animals ON animals.owner_id=owners.id;
+
+--How many animals are there per species?
+SELECT species.name, COUNT(*) from animals JOIN species ON animals.species_id=species.id GROUP BY species.name;
+
+--List all Digimon owned by Jennifer Orwell.
+SELECT animals.name FROM animals JOIN owners ON animals.owner_id=owners.id WHERE owners.full_name='Jennifer Orwell' AND species_id=2;
+
+--List all animals owned by Dean Winchester that haven't tried to escape.
+SELECT animals.name FROM animals JOIN owners ON animals.owner_id=owners.id
+WHERE owners.full_name='Dean Winchester' AND escape_attempts=0;
+
+--Who owns the most animals?
+SELECT owners.full_name, COUNT(animals.name) AS total FROM owners LEFT JOIN animals ON animals.owner_id=owners.id
+GROUP BY owners.full_name ORDER BY total DESC LIMIT 1;
+
+/*Queries to answer the following questions asked in the project requirements: */
+
+--Who was the last animal seen by William Tatcher?
+SELECT vets.name, animals.name, date_of_visit FROM vets JOIN visits ON vets.id=visits.vets_id
+JOIN animals ON animals.id= visits.animals_id WHERE vets.name ='William Tatcher' ORDER BY visits.date_of_visit DESC LIMIT 1;
+
+--How many different animals did Stephanie Mendez see?
+SELECT COUNT(*) as total_animals from vets JOIN visits ON vets.id = visits.vets_id WHERE name='Stephanie Mendez';
```

Comment:
- [ ] To get the number of different animals Stephanie Mendez saw there should be a join to the animal's table also. e.g.
```sql
SELECT DISTINCT animals.name
FROM visits
LEFT JOIN animals ON animals.id = visits.animal_id
LEFT JOIN vets ON vets.id = visits.vet_id
WHERE vets.name = 'Stephanie Mendez';
```